common/numerated/src/tree.rs

```diff
@@ -255,21 +253,24 @@ impl<T: Numerated> IntervalsTree<T> {
         }
     }
 
-    /// The same as [`Self::insert`], but returns [`I::Error`] if `try_into` [Interval] fails.
-    pub fn try_insert<I: TryInto<Interval<T>>>(&mut self, interval: I) -> Result<(), I::Error> {
-        let interval: Interval<T> = interval.try_into()?;
+    /// The same as [`Self::insert`], but returns [`I::Error`] if `try_into` [IntervalIterator] fails.
+    pub fn try_insert<I: TryInto<IntervalIterator<T>>>(
+        &mut self,
+        interval: I,
+    ) -> Result<(), I::Error> {
+        let interval: IntervalIterator<T> = interval.try_into()?;
         self.insert(interval);
         Ok(())
     }
 
     /// Remove `interval` from tree.
     /// - if `interval` is empty, then nothing will be removed.
-    /// - if `interval` is not empty, then after removing for each `p` ∈ `interval` ⇒ `p` ∉ `self`.
+    /// - if `interval` is not empty, then after removing: for each `p` ∈ `interval` ⇒ `p` ∉ `self`.
     ///
     /// Complexity: `O(log(n) + m)`, where
     /// - `n` is amount of intervals in `self`
     /// - `m` is amount of intervals in `self` ⋂ `interval`
-    pub fn remove<I: Into<Interval<T>>>(&mut self, interval: I) {
+    pub fn remove<I: Into<IntervalIterator<T>>>(&mut self, interval: I) {
         let Some((start, end)) = Self::into_start_end(interval) else {
             // Empty interval - nothing to remove.
             return;
```

Comment:
https://github.com/gear-tech/gear/blob/1128b5cc172e6fe004eded3a34d47782f304f849/common/numerated/src/tree.rs#L285-L287
its some kind of shortcut/fast-check, consider adding the same from `iter.start()`

Reply:
https://github.com/gear-tech/gear/blob/1128b5cc172e6fe004eded3a34d47782f304f849/common/numerated/src/tree.rs#L290-L291
maybe you could `.fold()` it and similar places?

Reply:
https://github.com/gear-tech/gear/blob/1128b5cc172e6fe004eded3a34d47782f304f849/common/numerated/src/tree.rs#L292-L296
wouldn't it be much easier and better introducing Ord for interval and fn like `interval1.contains(interval2)`?

Reply:
Consider `is_subset` for interval

Reply:
> maybe you could .fold() it and similar places?

I'd leave it as is for simplicity

Reply:
> https://github.com/gear-tech/gear/blob/1128b5cc172e6fe004eded3a34d47782f304f849/common/numerated/src/tree.rs#L285-L287
> 
> 
> its some kind of shortcut/fast-check, consider adding the same from `iter.start()`

What is `iter.start()`?

Reply:
> https://github.com/gear-tech/gear/blob/1128b5cc172e6fe004eded3a34d47782f304f849/common/numerated/src/tree.rs#L292-L296
> 
> 
> wouldn't it be much easier and better introducing Ord for interval and fn like `interval1.contains(interval2)`?

How exactly you want this contains must work `interval1.contains(interval2)` ?  in this code fragment is not a check for `contains` it is partially intersection. Also we use both bool statements: one for break and second for `left_interval` initialisation